sycl/include/CL/sycl/detail/cg_types.hpp

```diff
@@ -203,6 +203,12 @@ class HostKernel : public HostKernelBase {
   template <class ArgT = KernelArgType>
   typename detail::enable_if_t<std::is_same<ArgT, sycl::id<Dims>>::value>
   runOnHost(const NDRDescT &NDRDesc) {
+    using NameT =
+        typename detail::get_kernel_name_t<KernelName, KernelType>::name;
+    std::string KName = typeid(NameT *).name();
+    using KI = detail::KernelInfo<KernelName>;
+    constexpr bool StoreLocation = KI::callsAnyThisFreeFunction();
+
```

Comment:
These 5 lines of code seem to be copy-pasted 4 times. Can it be moved into a function returning constexpr bool ?

Reply:
Good catch.
We won't need the NameT and the KName anymore. I deleted them, and now it's simply the `using KI` and the call to the constexpr function.